plugins/woocommerce/client/blocks/assets/js/base/context/hooks/use-validation.ts

```diff
@@ -34,24 +34,19 @@ export const useValidation = (): ValidationData => {
 
 	const prefix = 'extensions-errors';
 
-	const {
-		hasValidationErrors,
-		getValidationError,
-	}: {
-		hasValidationErrors: boolean;
-		getValidationError: (
-			validationErrorId: string
-		) => ValidationContextError;
-	} = useSelect(
-		( select ) => {
+	const { hasValidationErrors } = useSelect( ( selectStore ) => {
+		const store = selectStore( validationStore );
+		return {
+			hasValidationErrors: store.hasValidationErrors(),
+		};
+	}, [] );
+
+	const getValidationError = useCallback(
+		( validationErrorId: string ) => {
 			const store = select( validationStore );
-			return {
-				hasValidationErrors: store.hasValidationErrors(),
-				getValidationError: ( validationErrorId: string ) =>
```

Comment:
A new function reference for `getValidationError` was created on every render, causing warnings.